### DISCUSSION TITLE
handle onError in next/image

### DISCUSSION THREAD

**Discussion by pepsdeveloper** [vercel/next.js#21921] [Ideas]
**Describe the feature you'd like to request**

I want to add an dummy image if the entered image src is not valid

**Describe the solution you'd like**

`<Image
  src={imagesrc}
  layout='fill'
  onError={handleError}
/>`

in handleError function i want to set the dummy image and it should replace src

**Describe alternatives you've considered**

Entered image is not valid then require any option to replace image src with dummy image

In my case i have added domain in next.config.js

`module.exports = {
    images: {
      domains:['images.example.com']
    }
  }`

but i get dynamic data from backend so some times i get wrong url

https://images.travelxp.com/somerror

so in this case i want to set the dummy image

**Comment by michrome** [comment]
What would you expect `next/image` to do if both the `src` and the dummy/fallback URLs were incorrect?

Personally, I think it would be preferable for the backend to return a dummy/fallback image if the src URL was incorrect.

**Reply by pepsdeveloper** [reply]
no @michrome i am expecting that to set a fallback image if the original image has any error like the below screenshot

![Screenshot 2021-02-08 at 11 21 34 AM](https://user-images.githubusercontent.com/76870082/107180926-cfdee380-69ff-11eb-9193-72da802f439e.png)

**Reply by pepsdeveloper** [reply]
Hi @michrome i got the solution for this

here is how i am doing.

`
export default class ProgressiveNextImage extends React.PureComponent {
constructor(props){
        super(props);
        this.state={
            isError: false
        }
        this.onError = this.onError.bind(this);
    }

onError(){
        this.setState({
            isError: true
        })
    }

 render(){
  const { imagesrc, alt } = this.props
 return (
   <Image
      src={this.state.isError ? "https://images.example.com/images/txpin/vector/general/errorimage.svg" : imagesrc}
      alt={alt}
      layout='fill'
      onError={this.onError}
      className='nextimage'
  />
)
}
}
`

**Reply by ryparker** [reply]
https://docs.github.com/en/github/writing-on-github/working-with-advanced-formatting/creating-and-highlighting-code-blocks

**Reply by hakuna0829** [reply]
@pepsdeveloper 
This solution works well. Btw onError() doesn't work with Storybook at all.
Can you check it there?
Thanks

**Reply by xiaoxiaosaohuo** [reply]
onError doesn't work at all

**Comment by hakuna0829** [comment]
Hi, I found a solution with Storybook, There is another prop called onErrorCapture similar onError and it works for me.
I hope this gives helps you guys. :)

**Reply by RizqiSyahrendra** [reply]
Thank you
